### DISCUSSION TITLE
Generating new API Keys

### DISCUSSION THREAD

**Discussion by inbytesdevteam** [supabase/supabase#5394] [Questions] [answered]
Hi, any idea on how to generate new API Keys? 

Let say you want to generate new API keys from time to time to make sure that your db is safe or your service role key got accidentally exposed to public and you want to generate a new key.

How would you do this? Thanks.

**Comment by GaryAustin1** [comment]
Currently you would have to ask support for a service_role key change.

**Reply by inbytesdevteam** [reply]
Thanks @GaryAustin1. Any idea when will the team implement this in the dashboard? :)

**Reply by burggraf** [reply]
This can be done here:  https://app.supabase.io/project/_/settings/api
When you generate a new JWT secret, it also generates new anon and service role keys.

**Comment by darora** [comment, marked as answer]
(Copying over Mark's comment so it can be marked as the answer)

This can be done here: [app.supabase.io/project/_/settings/api](https://app.supabase.io/project/_/settings/api)
When you generate a new JWT secret, it also generates new anon and service role keys; the older ones will no longer be valid.
